src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -197,15 +198,38 @@ public void AddRestore(string projectUniqueName)
         }
 
         public void AddProject(PackageSpec projectSpec)
-        {
+        {        
             // Find the unique name in the spec, otherwise generate a new one.
             var projectUniqueName = projectSpec.RestoreMetadata?.ProjectUniqueName
                 ?? Guid.NewGuid().ToString();
 
             if (!_projects.ContainsKey(projectUniqueName))
             {
-                _projects.Add(projectUniqueName, projectSpec);
+                var projectToRestore = projectSpec;
+                if (projectSpec.RestoreMetadata.CentralPackageVersionsEnabled)
```

Comment:
Can you elaborate why this is necessary? 
I thought we moved it down to where the original project spec is created.

Reply:
If here the entrance points(StaticGraphRestore etc) will only collect the data. I think the in this way it will be easier to maintain and reduce the chance to miss an entry point.

Reply:
It does something unobvious. 
Like the method name is add project but it does more, persoally I find that more confusing. thoughts @NuGet/nuget-client

Reply:
As discussed added https://github.com/NuGet/Home/issues/9175